final/MealsharingNET/Controllers/MealController.cs

```diff
@@ -0,0 +1,48 @@
+using Microsoft.AspNetCore.Mvc;
+using MealsharingNET.Models;
+
+
+
+namespace MealsharingNET.Controllers;
+
+[ApiController]
+[Route("meals")]
+public class MealController : ControllerBase
+{
+    private IMealRepository _repo;
+
+    public MealController(IMealRepository repo)
+    {
+        _repo = repo;
+    }
+
+    [HttpGet("List")]
+    public async Task<ActionResult<IEnumerable<Meal>>> ListAllMeals()
+    {
+        var response = await _repo.ListMeals();
+        return new OkObjectResult(response);
+    }
+    [HttpPost("Add")]
+    public void AddMeal([FromBody] Meal m)
+    {
+        _repo.Add(m);
```

Comment:
If this is a Task in the Interface, then you should await it and make the method async Task.

`public async Task ...`
`await _repo ...`